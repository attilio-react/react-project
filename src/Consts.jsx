const consts = { 
    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1
}

export {consts}
