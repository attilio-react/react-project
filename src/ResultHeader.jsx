import React from 'react';

import {SearchContext} from './SearchContext.jsx'

import {Label} from './common/Label.jsx'

class ResultHeader extends React.PureComponent {
  render () {
    return <SearchContext.Consumer> 
          {ctx => (
              <div>
               <Label text={`${ctx.total} movies found`} />  
              </div>
          )}
          </SearchContext.Consumer>
  }
}


export {ResultHeader}


