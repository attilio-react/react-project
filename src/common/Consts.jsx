const consts = { 
    SEARCH_SCREEN : 0,
    DETAIL_SCREEN: 1,


    SEARCH_BY_TITLE : 0,
    SEARCH_BY_GENRE : 1,

    SORT_BY_RELEASE_DATE : 0,
    SORT_BY_RATING : 1,


    PARAM_SORT_BY: 'sortBy',
    PARAM_SORT_ORDER: 'sortOrder',
    PARAM_SEARCH: 'search',
    PARAM_SEARCH_BY: 'searchBy',

    VALUE_BY_GENRE: 'genres',
    VALUE_BY_TITLE: 'title',
    VALUE_BY_RELEASE_DATE: 'release_date',
    VALUE_BY_VOTE_AVERAGE: 'vote_average',

    VALUE_ASC: 'asc',
    VALUE_DESC: 'desc'
}

export {consts}
