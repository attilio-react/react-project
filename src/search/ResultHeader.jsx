import React from 'react';

import {consts} from 'Common/Consts.jsx'

import {SearchContext} from './SearchContext.jsx'

import {Button} from 'Common/Button.jsx'
import {Label} from 'Common/Label.jsx'

class ResultHeader extends React.PureComponent {
  render () {
    return <SearchContext.Consumer> 
          {ctx => {
              const {total, releaseDateClickCb, ratingClickCb, sortBy} = ctx,
                  releaseDateSelected = (sortBy === consts.SORT_BY_RELEASE_DATE),
                  ratingSelected = !releaseDateSelected
              return <div>
               <Label text={`${total} movies found`} />  
               &nbsp;
               &nbsp;
               &nbsp;
               &nbsp;
               <Label text='Sort by: '/ >
               &nbsp;
               <Button onClick={releaseDateClickCb} caption='release date' selected={releaseDateSelected} />
               &nbsp;
               <Button onClick={ratingClickCb} caption='rating' selected={ratingSelected}/>
              </div>
          }}
          </SearchContext.Consumer>
  }
}


export {ResultHeader}


