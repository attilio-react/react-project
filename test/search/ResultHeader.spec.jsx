import * as React from 'react';
import {mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {ResultHeader} from '|search/ResultHeader.jsx';
import {SearchContext} from '|search/SearchContext.jsx';
import {consts} from '|common/Consts.jsx'

const CONTEXT = {
    total: 50,
    releaseDateClickCb: jest.fn(),
    ratingClickCb: jest.fn()
}

describe('ResultHeader rendering', () => {
  it('Should render when sort by release date', () => {
    const 
        component = renderer.create(
            <SearchContext.Provider value={{...CONTEXT, sortBy: consts.SORT_BY_RELEASE_DATE}}>
               <ResultHeader />
            </SearchContext.Provider>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('Should render when sort by rating', () => {
    const 
        component = renderer.create(
            <SearchContext.Provider value={{...CONTEXT, sortBy: consts.SORT_BY_RATING}}>
               <ResultHeader />
            </SearchContext.Provider>),
        json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});


