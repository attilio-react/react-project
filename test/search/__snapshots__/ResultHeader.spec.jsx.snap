// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultHeader rendering Should render when sort by rating 1`] = `
<div>
  <span>
    50 movies found
  </span>
         
  <span>
    Sort by: 
  </span>
   
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
    type="button"
    value="release date"
  />
   
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "underline",
      }
    }
    type="button"
    value="rating"
  />
</div>
`;

exports[`ResultHeader rendering Should render when sort by release date 1`] = `
<div>
  <span>
    50 movies found
  </span>
         
  <span>
    Sort by: 
  </span>
   
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "underline",
      }
    }
    type="button"
    value="release date"
  />
   
  <input
    onClick={[MockFunction]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
    type="button"
    value="rating"
  />
</div>
`;
